text/0040-token-metadata.md

```diff
@@ -0,0 +1,167 @@
+[motivation]: #motivation
+
+[MCIP #25](https://github.com/mobilecoinfoundation/mcips/pull/0025) and [MCIP #37](https://github.com/mobilecoinfoundation/mcips/pull/0037)
+have specified ways for new tokens to exist on chain. However, for user-facing applications to make effective use of this feature,
+they need more information about a token than just a token id -- they ultimately need to know how they should present balance information
+to the user.
+
+One way to do this is to hard-code all metadata into clients, which might work initially. After this however, this will mean that new tokens
+that are created on chain can't actually be used by users until a client update has been released and the users have updated.
+
+(In the [MCIP #31](https://github.com/mobilecoinfoundation/mcips/pull/0025) implementation PR, we did add a proto enum called `KnownTokenIds`
+which can be shipped with clients, as a stop-gap measure.)
+
+If the clients can dynamically query this information, then a client update is not required for clients to make use of new tokens which are
+created, which significantly reduces the amount of time it takes for a new token to be available to users.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Token metadata includes the following:
+
+* Short (exchange ticker) name of a token, e.g. MOB. (This is not expected to be translated.)
+* Long (english) name of a token, e.g. MobileCoin. (This is expected to be translated, but the client must maintain the translation list.)
+* Display conversion factor, e.g. 10^12. This conversion factor converts from the smallest representable units (`u64`) which appear in the blockchain,
+  to the user-displayable unit. (Note that this is possibly not a power of 10.) `u64` amounts
+  which are obtained from `TxOut`'s on chain should be divided by this number, for the token id, before being displayed to the user.
+* (Optional) URL pointing to an SVG logo of the token. (This is expected to be scalable and appropriate for display on devices of many form factors.)
+* (Optional) URL pointing to an MCIP which proposed the token.
+
+The consensus network will provide an endpoint at which this metadata can be requested for one or more token ids.
+
+This data will be provided at startup time via a file to the consensus node. (In the future, we may consider to store this data on chain.)
+The node should require this data for any token which is configured to exist (has a minimum fee in `tokens.toml`).
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+A token metadata proto has these fields
+
+```
+message TokenMetadata {
+    // The short name is a standard ticker symbol for the token
+    String short_name = 1;
+    // The long name is the standard english name for the token
+    String long_name = 2;
+    // The conversion factor is an unsigned 64 bit number expressed as a decimal string
+    String display_conversion_factor = 3;
```

Comment:
```suggestion
    uint64 display_conversion_factor = 3;
```

Reply:
okay, makes sense

also it's worth discussing, maybe this should be just "number of decimals" since that's what everyone seems to do, when I was researching for prior art